src/extensions/OrderBook.sol

```diff
@@ -0,0 +1,42 @@
+pragma solidity ^0.4.13;
+
+// NOTE: We're using address[100] to get back orderIDs from the orderFetcher as Solidity currently lacks
+// the ability to receive dynamic arrays from external contract calls. The list gets shrunk to the desired size
+// before returning
```

Comment:
Oh, we should remove `getOrderIDs` from `OrderBook` then.  Consider adding to this comment a reminder for that being a potential solution.  I personally don't think any of our internal contracts (not counting extensions) should be returning arbitrary sized arrays.